FV3/gfsphysics/GFS_layer/GFS_diagnostics.F90

```diff
@@ -46,12 +46,358 @@ module GFS_diagnostics
+    Diag(index)%data(nb)%var3 => IntDiag(nb)%q_dt(:,:,3)
+  enddo
+
+  index = index + 1
+  Diag(index)%axes = 3
+  Diag(index)%name = 'tendency_of_specific_humidity_due_to_microphysics'
+  Diag(index)%desc = 'water vapor tendency due to microphysics'
+  Diag(index)%unit = 'kg/kg/s'
+  Diag(index)%mod_name = 'gfs_phys'
+  Diag(index)%coarse_graining_method = 'mass_weighted'
+  allocate (Diag(index)%data(nblks))
+  do nb = 1,nblks
+    Diag(index)%data(nb)%var3 => IntDiag(nb)%q_dt(:,:,4)
+  enddo
+
+  index = index + 1
+  Diag(index)%axes = 3
+  Diag(index)%name = 'tendency_of_specific_humidity_due_to_change_in_atmosphere_mass'
+  Diag(index)%desc = 'residual water vapor tendency'
+  Diag(index)%unit = 'kg/kg/s'
+  Diag(index)%mod_name = 'gfs_phys'
+  Diag(index)%coarse_graining_method = 'mass_weighted'
+  allocate (Diag(index)%data(nblks))
+  do nb = 1,nblks
+    Diag(index)%data(nb)%var3 => IntDiag(nb)%q_dt(:,:,5)
+  enddo
+
+  ! Vertically integrated instantaneous specific humidity tendency diagnostics
+  index = index + 1
+  Diag(index)%axes = 2
+  Diag(index)%name = 'vertically_integrated_tendency_of_specific_humidity_due_to_turbulence'
+  Diag(index)%desc = 'vertically integrated water vapor tendency due to turbulence scheme'
+  Diag(index)%unit = 'kg/m**2/s'
+  Diag(index)%mod_name = 'gfs_phys'
+  Diag(index)%coarse_graining_method = 'area_weighted'
+  allocate (Diag(index)%data(nblks))
+  do nb = 1,nblks
+    Diag(index)%data(nb)%var2 => IntDiag(nb)%q_dt_int(:,1)
+  enddo
+
+  index = index + 1
+  Diag(index)%axes = 2
+  Diag(index)%name = 'vertically_integrated_tendency_of_specific_humidity_due_to_deep_convection'
+  Diag(index)%desc = 'vertically integrated water vapor tendency due to deep convection'
+  Diag(index)%unit = 'kg/m**2/s'
+  Diag(index)%mod_name = 'gfs_phys'
+  Diag(index)%coarse_graining_method = 'area_weighted'
+  allocate (Diag(index)%data(nblks))
+  do nb = 1,nblks
+    Diag(index)%data(nb)%var2 => IntDiag(nb)%q_dt_int(:,2)
+  enddo
+
+  index = index + 1
+  Diag(index)%axes = 2
+  Diag(index)%name = 'vertically_integrated_tendency_of_specific_humidity_due_to_shallow_convection'
+  Diag(index)%desc = 'vertically integrated water vapor tendency due to shallow convection'
+  Diag(index)%unit = 'kg/m**2/s'
+  Diag(index)%mod_name = 'gfs_phys'
+  Diag(index)%coarse_graining_method = 'area_weighted'
+  allocate (Diag(index)%data(nblks))
+  do nb = 1,nblks
+    Diag(index)%data(nb)%var2 => IntDiag(nb)%q_dt_int(:,3)
+  enddo
+
+  index = index + 1
+  Diag(index)%axes = 2
+  Diag(index)%name = 'vertically_integrated_tendency_of_specific_humidity_due_to_microphysics'
+  Diag(index)%desc = 'vertically integrated water vapor tendency due to microphysics'
+  Diag(index)%unit = 'kg/m**2/s'
+  Diag(index)%mod_name = 'gfs_phys'
+  Diag(index)%coarse_graining_method = 'area_weighted'
+  allocate (Diag(index)%data(nblks))
+  do nb = 1,nblks
+    Diag(index)%data(nb)%var2 => IntDiag(nb)%q_dt_int(:,4)
+  enddo
+
+  index = index + 1
```

Comment:
awesome, so this configures all the metrics you might want to run through the diagnostic methods?

Reply:
Yup exactly.  I'm largely mirroring how this is done for the existing physics diagnostics.